src/components/MyRecipes.vue

```diff
@@ -1,291 +1,566 @@
+    UploadImage,
+    VueEditor,
+    FlagPicker,
+    ValidationProvider,
+    ValidationObserver,
+    CookbookSelector,
+  },
+
+  methods: {
+    toggleEditor() {
+      this.inEditMode = !this.inEditMode;
+    },
+
+    addField(value, field) {
+      field.push({ value: { name: "", unit: "", thumbnail: "", link: "" } })
+      // Clear the thumbnail data for all ingredients
+      this.ingredients.forEach((ingredient, index) => {
+        this.$delete(ingredient, 'imageResults');
+        this.$delete(ingredient, 'showImageResults');
+      });
+    },
+    removeField(index, field) {
+      if (index > 0) {
+        field.splice(index, 1);
+      }
+    },
+
+    selectThumbnail(index, thumbnailUrl) {
+      this.ingredients[index].thumbnail = thumbnailUrl.src.original;
     },
-    data() {
-        return {
-            inEditMode: true,
-            uploadMessageDescription: "Upload Recipe Cover Image",
-            imageDimensionMsg: "Image dimension for best results (1127 x 650px)",
-            acceptTypes: ".png",
-            recipeDescription: "",
-            ingredients: [{ name: "", unit: "", thumbnail: "", link: "" }],
-            customToolbar: [
-                [{ header: [false, 1, 2, 3, 4, 5, 6] }],
-                ["bold", "italic", "underline", "strike"], // toggled buttons
-                [
-                    { align: "" },
-                    { align: "center" },
-                    { align: "right" },
-                    { align: "justify" }
-                ],
-                ["blockquote", "code-block"],
-                [{ list: "ordered" }, { list: "bullet" }, { list: "check" }],
-                [{ indent: "-1" }, { indent: "+1" }], // outdent/indent
-                [{ color: [] }, { background: [] }], // dropdown with defaults from theme
-                ["clean"] // remove formatting button
-            ]
+
+    async searchIngredientImages(index) {
+      const ingredient = this.ingredients[index].name;
+      const unit = this.ingredients[index].unit;
+      // Remove the existing imageResults property
+      this.$delete(this.ingredients[index], 'imageResults');
+      this.$delete(this.ingredients[index], 'showImageResults');
+      // Check if both name and unit are filled before making the API call
+      if (ingredient && unit) {
+        try {
+          this.$set(this.ingredients[index], 'loading', true); // Set loading state to true
+          // Make the API call to fetch the ingredient thumbnails
+          const response = await this.fetch_ingredient_thumbnail(ingredient)
+          // Extract the thumbnail URLs from the API response
+          const imageResults = response;
+          // Update the corresponding ingredient object with the image results
+          this.$set(this.ingredients[index], 'imageResults', imageResults);
+          this.$set(this.ingredients[index], 'showImageResults', imageResults.length > 0);
+        } catch (error) {
+          console.log(error)
```

Comment:
we have a global error handler already, kindly always use that in place of console log statements